tasks/gulp/compile-assets.js

```diff
@@ -183,33 +184,61 @@ gulp.task('scss:compile', function (done) {
   done()
 })
 
+// Convert a string from dashed (e.g: character-count) to PascalCase (e.g: CharacterCount)
+function convertFileNameToModuleName (filename) {
+  // Make the first letter of the filename uppercase
+  const uppercase = filename[0].toUpperCase() + filename.slice(1).toLowerCase()
+
+  // Remove any dashes and make the first letter after each dash uppercase
+  const moduleName = uppercase.replace(/-([a-z])/gi, function (match, char, index, str) {
+    return char.toUpperCase()
+  })
+
+  return moduleName
+}
+
 // Compile js task for preview ----------
 // --------------------------------------
-gulp.task('js:compile', () => {
-  // for dist/ folder we only want compiled 'all.js' file
-  const srcFiles = isDist ? configPaths.src + 'all.js' : configPaths.src + '**/*.js'
-
-  return gulp.src([
-    srcFiles,
-    '!' + configPaths.src + '**/*.test.js'
-  ])
-    .pipe(rollup({
-      // Used to set the `window` global and UMD/AMD export name.
-      name: 'GOVUKFrontend',
-      // Legacy mode is required for IE8 support
-      legacy: true,
-      // UMD allows the published bundle to work in CommonJS and in the browser.
-      format: 'umd'
-    }))
-    .pipe(gulpif(isDist, uglify({
-      ie8: true
-    })))
-    .pipe(gulpif(isDist,
-      rename({
-        basename: 'govuk-frontend',
-        extname: '.min.js'
-      })
-    ))
-    .pipe(eol())
-    .pipe(gulp.dest(destinationPath))
+gulp.task('js:compile', (done) => {
+  // For dist/ folder we only want compiled 'all.js'
+  const fileLookup = isDist ? configPaths.src + 'all.js' : configPaths.src + '**/!(*.test).js'
+
+  // Perform a synchronous search and return an array of matching file names
+  const srcFiles = glob.sync(fileLookup)
+
+  srcFiles.forEach(function (file) {
```

Comment:
Yeah I'd like to see us using less gulp rather than more :/

Reply:
Yeah, I came across a few things when I was researching but most gulp plugins seem very unloved and outdated now. Coupled with the fact that I don't think we have great gulp knowledge on the team and there's probably other ways of doing things now, I'm trying to avoid plugging in any new gulp things.